frontend/src/pages/PipelineDetailsV1.tsx

```diff
@@ -260,7 +261,14 @@ const PipelineDetailsV1: React.FC<PipelineDetailsV1Props> = ({
         )}
         {selectedTab === 1 && !!templateString && (
           <div className={css.containerCss} data-testid={'spec-yaml'}>
-            <PipelineSpecTabContent templateString={templateString || ''} />
+            {/* The templateString is from getPipelineTemplated() and it's JSON-formatted.
+              In order to make consistent with v2 pipeline behavior, we decide to render
+              it in yaml format.
+              safeLoad() convert string to object first, and safeDump() changes the object
+              to yaml format which is also acceptable by <PipelineSpecTabContent> */}
+            <PipelineSpecTabContent
+              templateString={jsyaml.safeDump(jsyaml.safeLoad(templateString)) || ''}
```

Comment:
Looks like we need to change this part to include V1 template instead, how do you think? https://github.com/kubeflow/pipelines/blob/master/frontend/src/components/PipelineSpecTabContent.tsx#L32-L36